packages/outputarea/src/model.ts

```diff
@@ -526,6 +526,9 @@ namespace Private {
     if (text === undefined) {
       text = '';
     }
+    if (!(newText.includes('\b') || newText.includes('\r'))) {
+      return text + newText;
+    }
```

Comment:
That's a good fast-track. Further down I guess we can also make it jump to index of `\b` or `\r` rather than process character by character. I can work on that.

Reply:
Yes, that would be one of the possible further optimizations.